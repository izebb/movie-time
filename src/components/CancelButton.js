import React from 'react';
import styled from 'styled-components';

const Svg = styled.svg`
  width: 60px;
  height: 60px;
  fill: #fff;
`;

const Button = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
`;

export const CancelButton = ({ onClick }) => {
  return (
    <Button onClick={onClick}>
      <Svg viewBox="0 0 100 100" x="0px" y="0px">
        <path d="M74,71.21,52.83,50,74,28.79A2,2,0,1,0,71.21,26L50,47.17,28.79,26A2,2,0,0,0,26,28.79L47.17,50,26,71.21A2,2,0,1,0,28.79,74L50,52.83,71.21,74A2,2,0,1,0,74,71.21Z" />
      </Svg>
    </Button>
  );
};

CancelButton.defaultProps = {
  onClick: () => {}
};
