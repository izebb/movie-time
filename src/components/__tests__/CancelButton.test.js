import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CancelButton } from '../CancelButton';

describe('CancelButton', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<CancelButton />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call `onClick`  when button is clicked', () => {
    const onClick = jest.fn();
    const component = mount(<CancelButton onClick={onClick} />);

    component.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
