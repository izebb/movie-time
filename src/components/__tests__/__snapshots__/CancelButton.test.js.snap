// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton should render without crashing 1`] = `
<CancelButton__Button
  onClick={[Function]}
>
  <CancelButton__Svg
    viewBox="0 0 100 100"
    x="0px"
    y="0px"
  >
    <path
      d="M74,71.21,52.83,50,74,28.79A2,2,0,1,0,71.21,26L50,47.17,28.79,26A2,2,0,0,0,26,28.79L47.17,50,26,71.21A2,2,0,1,0,28.79,74L50,52.83,71.21,74A2,2,0,1,0,74,71.21Z"
    />
  </CancelButton__Svg>
</CancelButton__Button>
`;
